src/TurbulenceConvection/EDMF_functions.jl

```diff
@@ -735,10 +736,32 @@ function filter_updraft_vars(
     a_max = edmf.max_area
 
     @inbounds for i in 1:N_up
-        @. prog_up[i].ρarea = max(prog_up[i].ρarea, 0)
-        @. prog_up[i].ρaθ_liq_ice = max(prog_up[i].ρaθ_liq_ice, 0)
-        @. prog_up[i].ρaq_tot = max(prog_up[i].ρaq_tot, 0)
-        @. prog_up[i].ρarea = min(prog_up[i].ρarea, ρ_c * a_max)
+        @inbounds for k in real_center_indices(grid)
+            if prog_up[i].ρarea[k] < FT(0)
+                aux_bulk.filter_flag_1[k] = 1
+            else
+                aux_bulk.filter_flag_1[k] = 0
+            end
+            if prog_up[i].ρaθ_liq_ice[k] < FT(0)
+                aux_bulk.filter_flag_2[k] = 1
+            else
+                aux_bulk.filter_flag_2[k] = 0
+            end
+            if prog_up[i].ρaq_tot[k] < FT(0)
+                aux_bulk.filter_flag_3[k] = 1
+            else
+                aux_bulk.filter_flag_3[k] = 0
+            end
+            if prog_up[i].ρarea[k] > ρ_c[k] * a_max
+                aux_bulk.filter_flag_4[k] = 1
+            else
+                aux_bulk.filter_flag_4[k] = 0
+            end
+        end
```

Comment:
Can we write this as broadcast expressions instead? This relies on `getindex`, and will need to be changed.

Reply:
Will do!